### DISCUSSION TITLE
Allow a user to contact us via our configured email

### DISCUSSION THREAD

**Discussion by ayoahha** [requarks/wiki#5939] [Help / Questions]
Hello,
I am facing an issue that is blocking me.
**_I would like any user to send me an email in order for me to create them an account._**
However i do not like to put this email like this, i would prefer encoding it, so that bots cannot read simply this email address...

I tried to add html link :
> ul
>     li
>       a href="" data-part1="a" data-part2="b" data-part3="email" data-part4="com" Contact /a
>     /li
> /ul

with a script in Page Properties (between <script> tags) :

    const links = document.querySelectorAll("[data-part1][data-part2][data-part3][data-part4]");
    for (const link of links) {
      const attrs = link.dataset;
      link.setAttribute(
        "href",
        `mailto:${attrs.part1}.${attrs.part2}@${attrs.part3}.${attrs.part4}`
      );
    }

**BUT it is not working and i guess i am missing some things...**

My questions are :
- Is it really possible to do that ?
- And how can we obfuscate an email address ?

Thanks for your help
Btw wonderful app, congrats to devs

**Comment by NGPixel** [comment]
Possible? Yes. Your script would need to be in the `page-ready` event for it to work (see https://docs.requarks.io/troubleshooting#how-to-execute-javascript-code-on-page-load)

This is however overkill and not very effective. A simple text `someone [at] company [dot] com` in the page is usually the best way. Another solution is to use an actual 3rd party form service which would completely hide your email address.

**Reply by ayoahha** [reply]
Ho ! Thanks.
Edit1 : unfortunately it is not working in the page-ready callback
```
<script>  
  window.boot.register('page-ready', () => {
    const links = document.querySelectorAll("[data-part1][data-part2][data-part3][data-part4]");
    for (const link of links) {
      const attrs = link.dataset;
      link.setAttribute(
        "href",
        `mailto:${attrs.part1}.${attrs.part2}@${attrs.part3}.${attrs.part4}`
      );
    }
  })
</script>
```


A contact form will imply some backend or PHP stuffs no ?
Do you have an example ?
